File: 214/solution.py
maxm = 4000 * 10000 + 1
num = list(range(0, maxm))
e = [0] * maxm

for i in range(2, maxm):
    if num[i] == i:
        for j in range(i, maxm, i):
            num[j] = int(num[j] / i * (i - 1))
e[1] = 1
ans = 0
for i in range(2, maxm):
    e[i] = e[num[i]] + 1
    if e[i] == 25 and num[i] == i - 1:
        ans += i
print(ans)

File: 214/solution.go
package main

import "fmt"

var num, e []int
var maxm = 4000*10000 + 1

func init() {
	num = make([]int, maxm)
	for i := 1; i < maxm; i++ {
		num[i] = i
	}
	e = make([]int, maxm)
	for i := 2; i < maxm; i++ {
		if num[i] == i {
			for j := i; j < maxm; j += i {
				num[j] = num[j] / i * (i - 1)
			}
		}
	}
}

func main() {
	e[1] = 1
	ans := int64(0)
	for i := 2; i < maxm; i++ {
		e[i] = e[num[i]] + 1
		if e[i] == 25 && num[i] == i-1 {
			ans += int64(i)
		}
	}
	fmt.Printf("%d\n", ans)
}
